packages/worker/src/tasks/syncPackageVulnerabilities.ts

```diff
@@ -0,0 +1,140 @@
+import { fetchAdvisoryDbSnapshot } from '../githubAdvisoryDb/advisoryDb';
+import gunzip from 'gunzip-maybe';
+import TarStream from 'tar-stream';
+import Stream from 'node:stream';
+import { pipeline } from 'node:stream/promises';
+import {
+  OpenSourceVulnerability,
+  OSVAffectedRangeType,
+  PackageVulnerability,
+} from '@gradejs-public/shared';
+
+type PackageVulnerabilityData = Pick<
+  PackageVulnerability,
+  'packageName' | 'packageVersionRange' | 'osvId' | 'osvData'
+>;
+
+async function drainStreamToBuffer(stream: Stream.Readable): Promise<Buffer> {
+  return new Promise((resolve, reject) => {
+    const dataChunks: any[] = [];
+
+    stream.on('data', (chunk) => dataChunks.push(chunk));
+    stream.on('end', () => resolve(Buffer.concat(dataChunks)));
```

Comment:
Won't it leak memory here and below? Actually, I'm not sure about removing listeners from the stream here, but I thought it's worth mentioning it :)